tests/SAML2/AuthnRequestTest.php

```diff
@@ -676,8 +674,10 @@ public function testSettingNameIDPolicyToIncorrectTypeSPNameQualifier(): void
         $request->setDestination('https://tiqr.example.org/idp/profile/saml2/Redirect/SSO');
 
         // SPNameQualifier must be a string
-        $nameIdPolicy = ["SPNameQualifier" => true];
-        $this->expectException(\InvalidArgumentException::class, 'Invalid Argument type: "string" expected');
+        $nameIdPolicy = new NameIDPolicy();
+        $this->expectException(\TypeError::class);
```

Comment:
The only question I ask myself about this PR is whether it is still desirable to test for invalid argument types now that we have fully typehinted the codebase..

Reply:
I guess psalm would uncover these errors, so tests can then focus on uncovering business logic errors rather than programming ones...

Reply:
Exactly! Great! I'll remove those tests..  I don't think they help for coverage anyway, since the method called isn't run at all..